TeachingRecordSystem/src/TeachingRecordSystem.Core/Services/TrsDataSync/TrsDataSyncHelper.cs

```diff
@@ -1308,92 +1309,178 @@ public async Task<int> SyncEventsAsync(IReadOnlyCollection<dfeta_TRSEvent> event
         return events.Count;
     }
 
-    private async Task<int> SyncRoutesAsync(
-        IReadOnlyCollection<RouteToProfessionalStatus> routes,
+    private async Task<Guid[]> EnsureSyncPersonsAsync(Guid[] personIds, bool ignoreInvalid = false, bool dryRun = false, CancellationToken cancellationToken = default)
+    {
+        using var dbContext = TrsDbContext.Create(trsDbDataSource);
+        var alreadySyncedPersonIds = await dbContext.Persons
```

Comment:
Does this need `IgnoreQueryFilters()`?

Reply:
yes otherwise it doesn't sync any inactive contacts which aren't currently in TRS